### PR TITLE
Split partially bounded range builders out for low and high bounded ranges

### PR DESCRIPTION
**Pull request by ronawho**
Split `chpl_build_partially_bounded_range` into `chpl_build_low_bounded_range`
and `chpl_build_high_bounded_range`. This is in preparation for an optimization
of anonymous counted low bounded ranges (e.g. `for i in 1..#10`)

Since the bound type is now embedded in the name, the parser doesn't need to
pass the BoundRangeType to the builder. Splitting out the range builders also
allows us to clean up/improve the builder error messages, function signatures,
and calls to the range constructors.

This also includes some chpldoc bug fixes where range and domain literals were
displayed as the internal function calls instead of cleaned up source-like
code.

### CODE REVIEW

**Comment by ronawho** [comment]
@bradcray Peeled off of https://github.com/chapel-lang/chapel/pull/3154 because it became a bigger change than I wanted. It includes all the changes you mentioned in that PR

https://github.com/ronawho/chapel/commit/a0bcd47acced9c46b04f36573c17198903052b87 is the main commit

File: compiler/parser/chapel.ypp [hunk 1]
```diff
@@ -1433,11 +1433,11 @@ expr:
...
     { $$ = new CallExpr("chpl_build_bounded_range", $1, $3); }
 | expr TDOTDOT
-    { $$ = new CallExpr("chpl_build_partially_bounded_range", buildDotExpr("BoundedRangeType", "boundedLow"), $1); }
+    { $$ = new CallExpr("chpl_build_low_bounded_range", $1); }
 | TDOTDOT expr
-    { $$ = new CallExpr("chpl_build_partially_bounded_range", buildDotExpr("BoundedRangeType", "boundedHigh"), $2); }
+    { $$ = new CallExpr("chpl_build_high_bounded_range", $2); }
 | TDOTDOT
-    { $$ = new CallExpr("chpl_build_unbounded_range", buildDotExpr("BoundedRangeType", "boundedNone")); }
+    { $$ = new CallExpr("chpl_build_unbounded_range"); }
```
Comment (bradcray) [inline compiler/parser/chapel.ypp]: Nice cleanup, thanks!

**Comment by bradcray** [comment]
This looks good to me, thanks!  I was surprised that the flex output file changed when (it appeared) you didn't change chapel.lex.  Maybe just a different version of flex than the last person who ran it (?).  I don't think it's worth looking into, just noting it.

**Comment by ronawho** [comment]
Yeah, I was also surprised especially because the original PR didn't change the flex output. I think it's because Michael recently committed changes made with bison 2.5 and I used 3.0.4

**Comment by ronawho** [comment]
@lydia-duncan just a heads up that this resolves the `buildDomainExpr()` bug from #3167

**Comment by lydia-duncan** [comment]
:+1:

**Pull request merged**